### PR TITLE
Junos device type (config backup, NTP check)

### LINKED ISSUE #8: JunOS support
JunOS is basically FreeBSD and supports SFTP just fine. It even has a "SFTP push on commit" mechanism that we might want to implement in addition to doing pulls.

The remote control class should wrap https://github.com/Juniper/py-junos-eznc instead of doing pexpect-style automation.  

Free vMX trial = perfect dev environment, a Juniper employee even made Docker file (https://github.com/mwiget/vmxdocker)

Docs stub: https://github.com/leoluk/configmaster/blob/master/docs/devices/junos.rst

### PR DESCRIPTION
**Pull request by leoluk**
Fixes #8

### CODE REVIEW

File: configmaster/management/handlers/config_backup.py [hunk 1]
```diff
@@ -399,9 +403,13 @@ def run_completed(cls):
...
-            filename = os.path.join(
-                Repository.objects.get(id=1).path,
-                'Meta', "{}_filter.txt".format(
+
+            path = os.path.join(Repository.objects.get(id=1).path, 'Meta')
+
+            if not os.path.exists(path):
+                os.mkdir(path)
+
+            filename = os.path.join(path, "{}_filter.txt".format(
```
Comment (Alexander-N) [inline configmaster/management/handlers/config_backup.py]: This doesn't seem to be related to the Junos device type. I would prefer something like this to be in a separate commit, to make it easy to revert specific changes.

Comment (leoluk) [reply configmaster/management/handlers/config_backup.py]: A separate fix would be squashed on merge, and it's a trivial fix. I agree on principle, but let's not be more catholic than the pope ;-)

Comment (Alexander-N) [reply configmaster/management/handlers/config_backup.py]: Squashing all commits is not the only merge option. Regarding the catholisism: agreed, this is why it is already approved :-)

Comment (leoluk) [reply configmaster/management/handlers/config_backup.py]: Agreed then :-)

**Comment by leoluk** [comment]
Needs to be cherry-picked on top of master as soon as #37 is merged
